standardize graph output indentation with 2 spaces

rickypai opened the pull request:
Currently, when outputting graphs from Bazel with `bazel query --output=graph`, the indentations for different types of lines are inconsistent:

- `node` is indented with 2 spaces
- edges and vertices are not indented
- labels are indented with 1 space

Example using [rules_go](https://github.com/bazelbuild/rules_go):

```
$ bazel query "deps(//go/platform:darwin)" --output=graph
digraph mygraph {
  node [shape=box];
"//go/platform:darwin"
"//go/platform:darwin" -> "//go/toolchain:darwin"
"//go/toolchain:darwin"
"//go/toolchain:darwin" -> "@bazel_tools//platforms:osx"
"@bazel_tools//platforms:osx"
"@bazel_tools//platforms:osx" -> "@bazel_tools//platforms:os"
"@bazel_tools//platforms:os"
}
Loading: 0 packages loaded
```

It would be nice for the indentations to be present (for edges and vertices), and consistent.

As for the indentation scheme, it seems inconclusive as to which type the language prefers:

- the [DOT Language Spec](https://www.graphviz.org/doc/info/lang.html) does not really specify tabs or spaces, but the examples used either contained no indentation (!) or one with 2 spaces
- the official [examples](https://www.graphviz.org/gallery/) contain samples of no indentation, tabs, or 2 spaces, with tabs being most popular
- the [Wikipedia page](https://en.wikipedia.org/wiki/DOT_(graph_description_language)) uses tabs

I'm using 2 spaces to conform with skylark. This is the result from the same query with this PR:

```plaintext
$ /Users/ricky/workspace/bazel/bazel-bin/src/bazel query "deps(//go/platform:darwin)" --output=graph
Starting local Bazel server and connecting to it...
DEBUG: /private/var/tmp/_bazel_ricky/5c63e3be3c60ec773878bdf16f25adcc/external/bazel_toolchains/rules/version_check.bzl:45:9: 
Current running Bazel is not a release version and one was not defined explicitly in rbe_autoconfig target. Falling back to '0.23.0'
digraph mygraph {
  node [shape=box];
  "//go/platform:darwin"
  "//go/platform:darwin" -> "//go/toolchain:darwin"
  "//go/toolchain:darwin"
  "//go/toolchain:darwin" -> "@bazel_tools//platforms:osx"
  "@bazel_tools//platforms:osx"
  "@bazel_tools//platforms:osx" -> "@bazel_tools//platforms:os"
  "@bazel_tools//platforms:os"
}
Loading: 3 packages loaded
```

gregestren commented:
Nice catch! And thanks for such a thorough analysis.

I'd stick with two spaces, which fits closer to [style guide recommendations](https://docs.bazel.build/versions/master/skylark/bzl-style.html#python-style) although I acknowledge this isn't Starlark. I think of Bazel as a space-oriented system in general.

rickypai commented:
@gregestren thanks a lot for the review. I rebased master due to merge conflicts and switched to 2 spaces in https://github.com/bazelbuild/bazel/pull/8007/commits/a1e69f12e58eb4499901152f7aba6aa99609bd83

let me know if there are other concerns.

gregestren commented:
One more quick request before we merge: could you update the PR description to mention that you're using spaces now and maybe throw out an "after" example at the end of what it the sample output looks like with your change?

rickypai commented:
sure let me do that real quick

rickypai commented:
@gregestren updated the PR description and posted after result

gregestren commented:
Thank you!

gregestren commented:
Just FYI I'm trying to merge this but running into a silly complication with the merge / export script. Need a bit of patience!

rickypai commented:
no problem and no rush. let me know if there's any way i could help.

gregestren commented:
Update: still working through this. We still have a few query tests inside Google that haven't yet been exported to GitHub (I'm not aware of any reason why they couldn't be, so consider that a TODO).  They embed expectations about the spacing which I think are just test brittleness, nothing wrong with what you're doing. But they highlight the possibility that people have tools/scripts that depend on the current formatting.

My current position is to still push this forward and if we find such cases then ask people to update their scripts. I don't believe this formatting is a guarantee of the Bazel user contract.

rickypai commented:
Thanks a lot!
